File: spring-multisport/src/main/java/com/demo/multisport/entities/page/Article.java
package com.demo.multisport.entities.page;


import lombok.*;
import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;
import java.util.Set;


@Entity
@Table(name = "articles")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @NotBlank
    @Size(min = 4, max = 50)
    @Column(length = 50, nullable = false, unique = true)
    private String title;

    @NotBlank
    @Column(columnDefinition = "TEXT NOT NULL")
    private String content;

    @NonNull
    @Column(name = "published_at", columnDefinition = "DATETIME DEFAULT NOW() NOT NULL")
    private LocalDateTime publishedAt;

    @NotBlank
    @NonNull
    @Column(name = "path_file")
    private String pathFile;

    @OneToMany(mappedBy = "article")
    private Set<Comment> comments;

    public Article(@NonNull String title,@NonNull String content, Set<Comment> comments) {
            this.title = title;
            this.content = content;
            this.comments = comments;
            this.publishedAt = LocalDateTime.now();
    }

    public Article(@NonNull String title,@NonNull String content, Set<Comment> comments,@NonNull LocalDateTime publishedAt) {
        this.title = title;
        this.content = content;
        this.comments = comments;
        this.publishedAt = publishedAt;
    }

    public Article withPublishedAt() {
        this.publishedAt = LocalDateTime.now();
        return this;
    }
}

File: angular-multisport/src/app/models/page/Article.ts
import { Comment } from "./Comment"

export class Article {
    private _title: string
    private _content: string
    private _pictureBase64: string
    private _publishedAt: Date
    private _comments: Comment[]

    constructor() {
        this._title = ''
        this._content = ''
        this._pictureBase64 = ''
        this._publishedAt = new Date()
        this._comments = []
    }

    withTitle(title: string): Article {
        this._title = title
        return this
    }

    withPublishedAt(publishedAt: Date) {
        this._publishedAt = publishedAt
        return this
    }

    withContent(content: string): Article {
        this._content = content
        return this
    }

    withPictureBase64(pictureBase64: string): Article {
        this._pictureBase64 = pictureBase64
        return this
    }

    withComments(comments: Comment[]): Article {
        this._comments = comments
        return this
    }

    get title(): string {
        return this._title
    }

    get content(): string {
        return this._content
    }

    get pictureBase64() : string {
        return this._pictureBase64
    }

    get publishedAt(): Date {
        return this._publishedAt
    }

    get comments(): Comment[] {
        return this._comments
    }
}